code/game/atoms.dm

```diff
@@ -280,6 +280,13 @@ var/list/blood_splatter_icons = list()
 		var/obj/effect/decal/cleanable/vomit/this = new /obj/effect/decal/cleanable/vomit(src)
 		if(M.reagents)
 			M.reagents.trans_to(this, M.reagents.total_volume / 10)
+			for(var/datum/reagent/R in reagents.reagent_list)		//clears the stomach of anything that might be digested as food

```

Comment:
also the reagents list only contains... reagents. so this should be an istypeless for loop.